src/NLog/Internal/SortHelpers.cs

```diff
@@ -61,14 +62,14 @@ internal static class SortHelpers
         /// <returns>
         /// Dictionary where keys are unique input keys, and values are lists of <see cref="AsyncLogEventInfo"/>.
         /// </returns>
-        public static Dictionary<TKey, List<TValue>> BucketSort<TValue, TKey>(this IEnumerable<TValue> inputs, KeySelector<TValue, TKey> keySelector)
+        public static IDictionary<TKey, IList<TValue>> BucketSort<TValue, TKey>(this IEnumerable<TValue> inputs, KeySelector<TValue, TKey> keySelector)
```

Comment:
AFAIK the usage of interfaces is always slower than the direct type. Not sure if this is also for these cases.